codec-http2/src/main/java/io/netty/handler/codec/http2/HttpUtil.java

```diff
@@ -302,7 +303,11 @@ public static Http2Headers toHttp2Headers(FullHttpMessage in) throws Exception {
             HttpResponse response = (HttpResponse) in;
             out.status(new AsciiString(Integer.toString(response.status().code())));
         }
+        return out;
+    }
 
+    public static Http2Headers toHttp2HeadersWithoutReserved(HttpHeaders inHeaders) throws Exception {
```

Comment:
Actually the blacklist contains more than pseudo headers ... maybe "toHttp2HeadersWithoutBlacklisted" or `toFilteredHttp2Headers`?